CATROID-889 testJoinFunction added using Catrobat Language Test

Bhavya-22 opened the pull request:
Tested the joinFunction() using Formula editor by Pocket Code
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

wslany reviewed (changes requested):
Thanks!! 

But please use only a single "Finish tests" brick. The three scripts are executed in parallel, and because of the three "Finish tests" there's a race condition, and thus it is not guaranteed that all of the Asserts may be tested. Maybe make a single script out of it? Or coordinate with broadcast bricks? But I think in this case, a single script is the easiest.

Bhavya-22 commented:
@wslany 
Sir I have done the requested changes in the specified file

wslany reviewed (changes requested):
Why the change in the .idea file? I think that's not needed, right?

And please squash your PR into one commit.

Please also add one more Assert brick before the finish test to check the implicit conversion between numbers and strings, e.g.:
![image](https://user-images.githubusercontent.com/1747888/107116714-78147100-6875-11eb-9301-9279c3a8d24b.png)

Bhavya-22 commented:
@wslany  
Sir to be honest I had not change anything in .idea file all the thing that I changed that in the respective folder i just deleted by previous file and updated it with the new one. Sir, I don't how .idea file got changed

wslany commented:
> @wslany
> Sir to be honest I had not change anything in .idea file all the thing that I changed that in the respective folder i just deleted by previous file and updated it with the new one. Sir, I don't how .idea file got changed

Sure, just take this change out of the PR please.

Bhavya-22 commented:
@wslany 
Sir, all changes made as requested please let me know if there is something else

Bhavya-22 commented:
@wslany Sir is there any error remaining because of which my PR is still open
If so then please let me know

reichli reviewed (changes requested):
Hello @Bhavya-22 !

Mr. Slany asked me to have a look at your PR.
Your language test looks very good already!

I just found two more kinds of tests which would make sense in this language test:

1. test that empty strings are basically ignored and do not add any extra spaces, etc.
2. test that integers and strings can be mixed arbitrarily.

![Screenshot from 2021-06-06 22-37-58](https://user-images.githubusercontent.com/23620339/120939420-496db080-c718-11eb-970b-334454855a68.png)


Please add tests for this and test it in combinations like you did for your 'Number with spacing' tests.
That way we make sure that those tests work not only for the 1st argument, but also for the 2nd and so on.

Thank you!

Bhavya-22 commented:
Sure sir I will add this brick to with switching both arguments within each
other to check both argument functionality

On Mon, Jun 7, 2021, 2:17 AM Thomas Reichel ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hello @Bhavya-22 <https://github.com/Bhavya-22> !
>
> Mr. Slany asked me to have a look at your PR.
> Your language test looks very good already!
>
> I just found two more kinds of tests which would make sense in this
> language test:
>
>    1. test that empty strings are basically ignored and do not add any
>    extra spaces, etc.
>    2. test that integers and strings can be mixed arbitrarily.
>
> [image: Screenshot from 2021-06-06 22-37-58]
> <https://user-images.githubusercontent.com/23620339/120939420-496db080-c718-11eb-970b-334454855a68.png>
>
> Please add tests for this and test it in combinations like you did for
> your 'Number with spacing' tests.
> That way we make sure that those tests work not only for the 1st argument,
> but also for the 2nd and so on.
>
> Thank you!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Catrobat/Catroid/pull/3940#pullrequestreview-676901786>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANBECJ3FZ4LX43AXCL2BKWTTRPNGFANCNFSM4WCLHCHA>
> .
>

Bhavya-22 commented:
Firstly I am very sorry for not being able to contribute in these past few months due to some family emergencies.
But now I am able to contribute again and I added the test with bricks recommended by @reichli sir and also added a combination of that bricks please check it on Jira and let me know if any other updates are required.

Also @wslany please remove the previous tests ( 0,1,2 ) so that only the total modified version will remain or please give me the access to remove them.

Bhavya-22 commented:
Sir firstly sorry for the delay but I have added the bricks Thomas sir
mentioned along with one of combination. Please check the GitHub and Jira
and let me know about any further updates.


On Fri, Jun 11, 2021, 2:50 PM Bhavya Jain ***@***.***> wrote:

> Sure sir I will add this brick to with switching both arguments within
> each other to check both argument functionality
>
> On Mon, Jun 7, 2021, 2:17 AM Thomas Reichel ***@***.***>
> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> Hello @Bhavya-22 <https://github.com/Bhavya-22> !
>>
>> Mr. Slany asked me to have a look at your PR.
>> Your language test looks very good already!
>>
>> I just found two more kinds of tests which would make sense in this
>> language test:
>>
>>    1. test that empty strings are basically ignored and do not add any
>>    extra spaces, etc.
>>    2. test that integers and strings can be mixed arbitrarily.
>>
>> [image: Screenshot from 2021-06-06 22-37-58]
>> <https://user-images.githubusercontent.com/23620339/120939420-496db080-c718-11eb-970b-334454855a68.png>
>>
>> Please add tests for this and test it in combinations like you did for
>> your 'Number with spacing' tests.
>> That way we make sure that those tests work not only for the 1st
>> argument, but also for the 2nd and so on.
>>
>> Thank you!
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Catrobat/Catroid/pull/3940#pullrequestreview-676901786>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ANBECJ3FZ4LX43AXCL2BKWTTRPNGFANCNFSM4WCLHCHA>
>> .
>>
>

wslany commented:
@Bhavya-22 Thank you! Please just update the pull request by force-pushing the changed file(s) to this pull request, in such a way that there is a single commit in this pull request, and move the ticket towards "code review". Then let me know via email please.

wslany commented:
There is no need to delete the old tests attached to the jira ticket, but if you want  I could do it. However, only the files in the pull request are important.

Bhavya-22 commented:
@wslany sir, I added the complete file/test on this pull request with a single commit please review it and let me know anything further which I should do.
Sorry for the late reply

reichli reviewed (approved):
Hi @Bhavya-22 !

I've just had a new look at your updated CLT and it looks very good now!
Thank you very much for adding those extra tests :)

Bhavya-22 commented:
Thank you sir for reviewing and if all possible please update the jira and GitHub tags
